classes/Project/ProjectFactory.mjs

```diff
@@ -0,0 +1,86 @@
+import {Page} from "../Page/Page.mjs"
+import Project from "./Project.mjs"
+
+let err_out = Object.assign(new Error(), {
+  status: 500,
+  message: "Unknown Server error"
+})
+
+export default class ProjectFactory {
+  constructor(data) {
+    this.data = data
+  }
+
+  static async fetchManifest(url) { 
+    return fetch(url)
+      .then((response) => {
+        return response.json()
+      })
+      .catch((err) => {
+        err_out.status = err.status??404
+        err_out.message = err.message??"Manifest not found. Please check URL" 
+        console.log(err)
+        throw err_out
+      })
+  }
+
+  static async processManifest(manifest) {
```

Comment:
This name should be more descriptive, thinking about how it'll appear in code. 

```javascript
let ProjectForDB = await ProjectFactory.processManifest(manifest)
database.save(ProjectForDB, "Project")
```

Perhaps a name like `databaseObjectFromManifest(manifest)` or `asDBObject(manifest)` would be more logical an appendage to `ProjectFactory`?

Reply:
`ProjectFactory.DBObjectFromManifest(manifest)`

Reply:
You could also add a little documentation for this, as it is a slightly complex and rather critical thing to understand.